src/sync/incomingSyncUtils.js

```diff
@@ -608,18 +608,26 @@ export const createOrUpdateRecord = (database, settings, recordType, record) =>
       break;
     }
     case 'Transaction': {
-      if (record.store_ID !== settings.get(THIS_STORE_ID)) break; // Not for this store
+      // Only sync transactions for this store.
+      const { store_ID } = record;
+      if (store_ID !== settings.get(THIS_STORE_ID)) break;
+
+      // Do not sync transactions without valid entry date.
+      const entryDate = parseDate(record.entry_date);
+      if (entryDate == null) break;
```

Comment:
I think this is a dangerous path without logging and mSupply Hub to scream at us.

Reply:
This PR is a short term band aid for a bug which is breaking core functionality. We need to have a bigger discussion about how we handle violations of data constraints in sync, but right now this seems like the best solution to get things working again.

Reply:
just a quick one to be annoying - `===` over `==` ?

Reply:
i think it's better to set entryDate = confirmDate.

And yes, we need to make our sync code more resilient (by-pass possible breaks), with logging to hub or bug-snag.

Reply:
Ohh and something else to discuss in an issue re improving resilience of sync -> requesting resync of a table (i.e when options were added on primary, sites that were upgraded to VM had to resync the options)